src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java

```diff
@@ -294,7 +409,24 @@ public void checkCoupling() {
          */
         private boolean isSignificant(String candidateClassName) {
             return !excludedClasses.contains(candidateClassName)
-                    && !candidateClassName.startsWith("java.lang.");
+                    && !isFromExcludedPackage(candidateClassName);
+        }
+
+        /**
+         * Checks if given class should be ignored as it belongs to excluded package.
+         * @param candidateClassName class to check
+         * @return true if we should not count this class.
+         */
+        private boolean isFromExcludedPackage(String candidateClassName) {
+            String classNameWithPackage = candidateClassName;
+            if (!candidateClassName.contains(DOT)) {
+                classNameWithPackage = parentContext
+                    .getClassNameWithPackage(candidateClassName)
+                    .orElse(null);
+            }
+
+            return classNameWithPackage != null && excludedPackages.stream()
+                .anyMatch(classNameWithPackage::startsWith);
```

Comment:
> So if I say to exclude package a.b and we have an import from a.b.c package, it will be 
> excluded since you use startsWith right?

Exactly so.

> If we did do subpackages too, user couldn't use this option on a common package in their 
> project.

We should probably change the format of the `excludedPackage` element, so that `a.b` and `a.b.*` will exclude `a.b` only (without subpackages) and `a.b.**` will exclude all packages inside `a.b`. Or introduce `includedPackages` (with same problem) with higher priority than `exludedPackage`.

> Either way, we should have a test for this, if we don't already, and some documentation in 
> the xdoc explaining this behavior.

Waiting for the decision and will update.